consensus/state.go

```diff
@@ -1216,9 +1214,12 @@ func (cs *ConsensusState) finalizeCommit(height int) {
 	//	* Fire before persisting state, in ApplyBlock
 	//	* Fire on start up if we haven't written any new WAL msgs
 	//   Both options mean we may fire more than once. Is that fine ?
-	types.FireEventNewBlock(cs.evsw, types.EventDataNewBlock{block})
-	types.FireEventNewBlockHeader(cs.evsw, types.EventDataNewBlockHeader{block.Header})
-	eventCache.Flush()
+	cs.eventBus.PublishEventNewBlock(types.EventDataNewBlock{block})
+	cs.eventBus.PublishEventNewBlockHeader(types.EventDataNewBlockHeader{block.Header})
+	err = txEventBuffer.Flush()
+	if err != nil {
+		cs.Logger.Error("Failed to flush event buffer", "err", err)
```

Comment:
Did we mean to return after this point?

Reply:
not sure. plus the current impl never returns an error. I just thought it will be good to have the interface of eventHub to return an error in case impl will change.

Reply:
errors in this function are critical. notice all the panics at the start of this function. `.Flush` should be expected to not error, but it's not critical, so long as nothing relies on it other than the websockets. The current code looks fine (ie just log the error). But if one day Flush actually errors, we may need a better way to get that error back up to the rpc server ...